docs/proposed/config-split.md

```diff
@@ -0,0 +1,420 @@
+options that apply to the whole code base (TLS, x509, crypto and tests) mostly
+related to the platform abstraction layer and testing. In tf_psa_crypto_config.h
+these configurations options are organized into two sections, one for the
+platform abstraction layer options and one for the others, respectively named
+"Platform abstraction layer" and "General and test configuration options".
+
+Then, the "Cryptographic mechanism selection (PSA API)" section is the
+equivalent of the pre-split crypto_config.h configuration file containing the
+PSA_WANT_ prefixed macros.
+
+The following section named "Cryptographic mechanism selection (extended API)"
+contains the configuration options for the cryptography mechanisms that are not
+yet part of the PSA cryptography API (like LMS or PK).
+
+It is followed by the "Data format support" section that contains configuration
+options of utilities related to various data formats (like base64 or ASN1 APIs).
+These utilities aim to facilitate the usage of the PSA cryptography API in other
+cryptography projects.
+
+Compared to Mbed TLS, the cryptography code in TF-PSA-Crypto is not located
+in a single directory but split between the PSA core (core directory) and the
+PSA builtin drivers (drivers/builtin/src directory). This is reflected in
+tf_psa_crypto_config.h with two sections respectively named "PSA core" and
+"Builtin drivers".
+
+Finally, the last section named "Legacy cryptography" contains the configuration
+options that will eventually be removed as duplicates of PSA_WANT_\* and
+MBEDTLS_PSA_ACCEL_\* configuration options.
+
+By contrast to mbedtls_config.h, tf_psa_crypto_config.h does not contain a
+section like the "Module configuration options" one containing non boolean
+configuration options. The configuration options that are not boolean are
+located in the same section as the boolean option they are associated to.
+
+Open question: do we group them into a subsection?
+
+## Repartition of the configuration options
+
+### In tf_psa_crypto_config.h, we have:
+* SECTION "Platform abstraction layer"
+#define MBEDTLS_HAVE_TIME
+#define MBEDTLS_HAVE_TIME_DATE
+//#define MBEDTLS_PLATFORM_MEMORY
+//#define MBEDTLS_PLATFORM_NO_STD_FUNCTIONS
+//#define MBEDTLS_PLATFORM_SETBUF_ALT
+//#define MBEDTLS_PLATFORM_EXIT_ALT
+//#define MBEDTLS_PLATFORM_TIME_ALT
+//#define MBEDTLS_PLATFORM_FPRINTF_ALT
+//#define MBEDTLS_PLATFORM_PRINTF_ALT
+//#define MBEDTLS_PLATFORM_SNPRINTF_ALT
+//#define MBEDTLS_PLATFORM_VSNPRINTF_ALT
+//#define MBEDTLS_PLATFORM_NV_SEED_ALT
+//#define MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT
+//#define MBEDTLS_PLATFORM_MS_TIME_ALT
+//#define MBEDTLS_PLATFORM_GMTIME_R_ALT
+//#define MBEDTLS_PLATFORM_ZEROIZE_ALT
+#define MBEDTLS_FS_IO
+//#define MBEDTLS_MEMORY_DEBUG
+//#define MBEDTLS_MEMORY_BACKTRACE
+//#define MBEDTLS_THREADING_ALT
+//#define MBEDTLS_THREADING_PTHREAD
+#define MBEDTLS_PLATFORM_C
+//#define MBEDTLS_THREADING_C
+#define MBEDTLS_TIMING_C
```

Comment:
I think that belongs in TLS: the only public functions there are not generic at all but made specifically for `mbedtls_ssl_set_timer_cb()`. So at this point really the module should be renamed to `ssl_timers.c` and the options accordingly. The renaming is out of scope here (again, 4.0?) but I think where this option goes is in scope.